### ISSUE TITLE
Category management dialog (was "Drop Down 'Pick' List For Categories not in A-Z order")

### ISSUE THREAD

**Issue by Oldbwl** [wedesoft/anymeal#32]
The Drop Down Category list order is probably in the order of creation by date at the moment -  so, when typing in the selection box a search item for Categories the list appears random. There are 2 useful reason for reordering in to A-Z
1. You know where to scroll to to get your item of preference and most importantly:
2.  You can see categories which are very similar in spelling which can result in totally different search results such as "Bell pepper - Green" or "Bell Pepper-Green" - An A-Z sort provides a means to see spelling errors etc. The alternative now is to Type in this Instance "B" then see every listed item to find pairs or more of categories meant to be the same scattered through the list. This is quite difficult.

**Comment by wedesoft** [comment]
The categories are ordered by number of recipes (I.e. popularity).

**Comment by Oldbwl** [comment]
OK, is there a config file where that can be altered?

**Comment by wedesoft** [comment]
I have created a [branch with that change](https://github.com/wedesoft/anymeal/compare/master...order-categories-by-name). It is not in a configuration file, so it needs rebuilding. Let me know if I should create a Windows installer or AppImage for you.

**Comment by Oldbwl** [comment]
I am a linux mint user, so will do a build from the code.

very grateful for this.

**Comment by wedesoft** [comment]
No problem. Happy to help. I guess if you don't have many recipes, sorting alphabetical is better. When you have 300000 recipes the Google approach of most frequent search term works better though, IMHO.

**Comment by Oldbwl** [comment]
Downloaded branch, been using for a week or so now, got rid of so many dupe categories with this in record time, really pleased and very grateful.

**Comment by wedesoft** [comment]
Ok, happy to hear that it is working for you :)

**Comment by wedesoft** [comment]
Hi, I have changed AnyMeal to order categories by recipe count descending first and by alphabet ascending second. So if you have lots of single recipe categories, they should now appear in alphabetical order.

**Comment by Oldbwl** [comment]
So will the following appear as shown below in example 1 or as in 2.
Example 1.
Cake
Dessert
Cakes
Desert
----the latter two having single entries

Example 2
Cake
Cakes
Desert
Dessert

I have like you in excess of 300K recipes, I have standardised the spelling of the categories to ensure all are the same major groupings, but of course with 350+ categories, I have made lots of spelling mistakes, and these are easy to catch in Example 2.

I type the first letter in the category search box and scroll down the list- - which can be very long, if all the single categories are at the end, then I am unlikely to notice the potential mistake, some are quite subtle, for example a capitalization of the word such as Baking Powder and baking powder, creates 2 categories, and I want to make them all the same - this is important when I export and import elsewhere.

Now if you created a Category edit/management  page? (lol)..... I ask for too much I know. LOL!!!!
